devito/ir/iet/utils.py

```diff
@@ -137,3 +141,30 @@ def derive_parameters(nodes, drop_locals=False):
         parameters = [p for p in parameters if not isinstance(p, LocalObject)]
 
     return parameters
+
+
+def find_offloadable_trees(iet):
+    """
+    Return the trees within ``iet`` that can be computed by YASK.
```

Comment:
this docstring needs to be updated

Reply:
Sure, my bad.

Reply:
""" 
    Return the trees within ``iet`` that can be considered offloadable.

    A tree is "offloadable" if it is embedded in a time stepping loop
    *and* all of the grids accessed by the enclosed equations are homogeneous
    (i.e., same dimensions and data type).
    
    Parameters
    ----------
    iet : list of :class:`Node`
        Original tree in which will be searched for the offloadable tree.
    
    Returns
    -------
    list of :class:`Node`
        A piece of the tree that contains a time stepping loop and all the 
        grid accessed are homogeneous.
    """

This is what i'll change it for.

Reply:
nitpicking:

I'm also dropping all "Return" ad the beginning of a docstring

How about something complete fresh, such as:
```
Find offloadable trees. A tree is offloadable when all of the array accesses are constant/affine functions of the Iteration variables and the Iteration bounds are fixed (but possibly symbolic).

Parameters
----------------
iet : :class:`Node`
    The searched tree

Returns
-----------
list of :class:`Node`
    Each item in the list is the root of an offloadable tree
```

Reply:
Agree, we have a `Returns` title so we could drop it in the method description.

That sounds simpler, i like it.